joomla/joomla-cms
# [4.0] Return sensible default from 'FileSystem - Local' plugin.

Pull Request for Issue #34358 .

### Summary of Changes

Changed the way directory settings are obtained in the `FileSystem - Local` plugin. If mo directory has been explicitly chosen yet, return Media Manager's 'Path to 'Path to Files Folder' setting.

### Testing Instructions

Make sure you have a fresh Joomla! 4.0 installation, or at least one with an unaltered `FileSystem - Local` plugin configuration.

Change both path's of the Media Manager configuration, i.e. 'Path to Files Folder' and 'Path to Images Folder'.

Open Media Manager.

**Beware of caching when fiddling with the Media Manager and its path settings!** (see also #34239)

### Actual result BEFORE applying this Pull Request

The result depends on proper configuration of the FileSystem - Local plugin. If no directory is chosen, which is the default, the plugin returns [{"directory": "images"}] and the Media Manager shows nothing.

### Expected result AFTER applying this Pull Request

A list of files in folder 'Path to Files Folder', if there are any, or an empty folder otherwise.

### Documentation Changes Required

It's probably a good thing to document this somewhere, preventing (new) users to get confused.
